test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -733,11 +733,15 @@ public void testEqualsAndHashcode() {
         }
     }
 
+    /**
+     * Generic test that checks that the toString method renders the XContent
+     * correctly.
+     */
     public void testToString() throws IOException {
         for (int runs = 0; runs < NUMBER_OF_TESTQUERIES; runs++) {
             QB testQuery = createTestQueryBuilder();
             XContentType xContentType = XContentType.JSON;
-            String toString = Strings.toString(testQuery);
+            String toString = randomBoolean() ? Strings.toString(testQuery) : testQuery.toString();
```

Comment:
To be clear, one thing is testing the toString output, and another thing is to make sure that the class implements the right interface, so that its output can be generated and is valid. I would do both from separate methods.